src/useQuery.ts

```diff
@@ -96,6 +96,8 @@ export function useQuery<
       ? 'cache-first'
       : actualCachePolicy;
 
+  const lastResult = useRef();
```

Comment:
we use this to be able to reference the **actual** previously returned data, we can't use `observableQuery.getLastResult()` because `observableQuery` may get invalidated and return `null`